cactus/plugin/builtin/pagecontext.py

```diff
@@ -0,0 +1,120 @@
+#coding:utf-8
+import logging
+import re
+
+
+logger = logging.getLogger(__name__)
+
+
+class PageContextPlugin(object):
+    """
+    A plugin to extract context variables from the top of the file
+
+    TODO:
+    - read style of metadata from config
+    - read splitChar from config
+    """
+
+    def preBuildPage(self, page, context, data):
+        """
+        Update the page context with the context variables from the file
+        and delete them
+        """
+        if not page.is_html() or not data:
+            return context, data
+
+        lines = data.splitlines()
+
+        # Look into the file to decide on the style of metadata
+        # TODO: make that configurable
+        if lines[0].strip() == u'---':
```

Comment:
The data can not be empty as of the current state of the code. Should I still test for it (`if not lines: return {}, []`)? I added a comment in the code to make it clear for future developers.